assets/main.js

```diff
@@ -0,0 +1,77 @@
+(function () {
+    var wrap = function (fn) {
+        return function () {
+            return fn.apply(arguments[0], Array.prototype.slice.call(arguments, 1));
+        }
+    };
+
+    var each = wrap(Array.prototype.forEach);
+    var filter = wrap(Array.prototype.filter);
+    var map = wrap(Array.prototype.map);
+    var mapNotUndef = function (obj, fn) {
+        return filter(map(obj, fn), function (it) {
+            return it !== undefined
+        });
+    };
```

Comment:
Not sure what happens here, but perhaps we can use `$.each`, `$.grep`, e.t.c here?